src/metabase/lib/drill_thru/underlying_records.cljc

```diff
@@ -116,6 +119,41 @@
    :row-count  row-count
    :table-name table-name})
 
+(mu/defn- filter-clauses-with-possible-bucketing :- [:sequential ::lib.schema.expression/boolean]
+  [original-column :- ::lib.schema.metadata/column
+   filter-column   :- ::lib.schema.metadata/column
+   value           :- :any]
+  (let [temporal-unit (or (::lib.underlying/temporal-unit original-column)
+                          (lib.temporal-bucket/raw-temporal-bucket original-column))]
+    (cond
+      ;; Ignore temporal bucketing if the `value` is NULL; use the `filter-column` directly.
+      (nil? value)
+      [(lib.filter/is-null filter-column)]
+
+      ;; If temporal-unit is given and it's a truncation unit, generate a `:between` clause
+      ;; for the right date(time) range.
+      (lib.schema.temporal-bucketing/datetime-truncation-units temporal-unit)
+      (let [[lo hi]   (u.time/to-range (u.time/coerce-to-timestamp value)
+                                       {:unit temporal-unit})
+            ;; If we're truncating to a date unit, truncate the endpoints to be just dates as well. Otherwise we'll
+            ;; get janky filter chip and widget, like `Created At: Week is May 12, 2024 00:00 - May 18, 2024 23:59`.
+            ;; With the endpoints turned to dates, it's `Created At: Week is May 12-18, 2024`.
+            base-type (if (lib.schema.temporal-bucketing/date-truncation-units temporal-unit)
+                        :type/Date
+                        :type/DateTime)]
+        [(lib.filter/between filter-column
+                             (u.time/format-for-base-type lo base-type)
+                             (u.time/format-for-base-type hi base-type))])
+
+      ;; Otherwise `temporal-unit` is an extraction unit, so compare the extracted (bucketed) column to the input
+      ;; `value` with `:=`; ie. `monthOfYear(column) = 3`.
+      temporal-unit
+      [(lib.filter/= (lib.temporal-bucket/with-temporal-bucket filter-column temporal-unit)
```

Comment:
The UI won't like this. Can you use `temporal-extract`-derived functions here? Like `get-month`